expression/testdata/partition_pruner_out.json

```diff
@@ -177,5 +177,757 @@
+        "Plan": [
+          "TableReader_7 2666.67 root partition:all data:Selection_6",
+          "└─Selection_6 2666.67 cop[tikv]  eq(test_partition.t2.a, test_partition.t2.id), gt(test_partition.t2.id, 10)",
+          "  └─TableFullScan_5 10000.00 cop[tikv] table:t2 keep order:false, stats:pseudo"
+        ]
+      },
+      {
+        "SQL": "select * from t3 where a=id and id >10",
+        "Result": null,
+        "Plan": [
+          "TableReader_7 2666.67 root partition:all data:Selection_6",
+          "└─Selection_6 2666.67 cop[tikv]  eq(test_partition.t3.a, test_partition.t3.id), gt(test_partition.t3.id, 10)",
+          "  └─TableFullScan_5 10000.00 cop[tikv] table:t3 keep order:false, stats:pseudo"
+        ]
+      },
+      {
+        "SQL": "select * from t1 where a=id and id >10 or a is null",
+        "Result": [
+          "<nil> <nil> <nil>"
+        ],
+        "Plan": [
+          "TableReader_7 2674.00 root partition:all data:Selection_6",
+          "└─Selection_6 2674.00 cop[tikv]  or(and(eq(test_partition.t1.a, test_partition.t1.id), gt(test_partition.t1.id, 10)), isnull(test_partition.t1.a))",
+          "  └─TableFullScan_5 10000.00 cop[tikv] table:t1 keep order:false, stats:pseudo"
+        ]
+      },
+      {
+        "SQL": "select * from t2 where a=id and id >10 or a is null",
+        "Result": [
+          "<nil> <nil> <nil>"
+        ],
+        "Plan": [
+          "TableReader_7 2674.00 root partition:all data:Selection_6",
+          "└─Selection_6 2674.00 cop[tikv]  or(and(eq(test_partition.t2.a, test_partition.t2.id), gt(test_partition.t2.id, 10)), isnull(test_partition.t2.a))",
+          "  └─TableFullScan_5 10000.00 cop[tikv] table:t2 keep order:false, stats:pseudo"
+        ]
+      },
+      {
+        "SQL": "select * from t3 where a=id and id >10 or a is null",
+        "Result": [
+          "<nil> <nil> <nil>"
+        ],
+        "Plan": [
+          "TableReader_7 2674.00 root partition:all data:Selection_6",
+          "└─Selection_6 2674.00 cop[tikv]  or(and(eq(test_partition.t3.a, test_partition.t3.id), gt(test_partition.t3.id, 10)), isnull(test_partition.t3.a))",
+          "  └─TableFullScan_5 10000.00 cop[tikv] table:t3 keep order:false, stats:pseudo"
+        ]
+      },
+      {
+        "SQL": "select * from t1 where a in (1,2) or a=4",
+        "Result": [
+          "1 1 1",
+          "2 2 2",
+          "4 4 4"
+        ],
+        "Plan": [
+          "TableReader_7 30.00 root partition:p0 data:Selection_6",
+          "└─Selection_6 30.00 cop[tikv]  or(in(test_partition.t1.a, 1, 2), eq(test_partition.t1.a, 4))",
+          "  └─TableFullScan_5 10000.00 cop[tikv] table:t1 keep order:false, stats:pseudo"
+        ]
+      },
+      {
+        "SQL": "select * from t2 where a in (1,2) or a=4",
+        "Result": [
+          "1 1 1",
+          "2 2 2",
+          "4 4 4"
+        ],
+        "Plan": [
+          "TableReader_7 30.00 root partition:all data:Selection_6",
```

Comment:
Why this is ALL instead of `p0,p1`?

Reply:
Because the partition expression of `t2` is `a*3 + b - 2*a - b`, and the condition only specified the column `a` value, doesn't specify the column `b` value, then we can't calculate the value from the partition expression, so can't to do prune partition here.